python/nano/src/bigdl/nano/pytorch/context_manager.py

```diff
@@ -46,12 +58,20 @@ class AutocastContextManager(BaseContextManager):
     This context manager is used for providing no grad and autocast context,
     which is used for bf16 model.
     """
-    def __init__(self, thread_num=None):
-        super().__init__(thread_num=thread_num)
+    def __init__(self, thread_num=None, accelerator=None):
+        super().__init__(thread_num=thread_num, accelerator=accelerator)
         self.autocast = torch.cpu.amp.autocast(enabled=True, dtype=torch.bfloat16)
 
     def __enter__(self):
         super().__enter__()
+        if self.accelerator == "jit":
+            if compare_version("torch", operator.le, "1.13.1"):
+                # onednn fusion for bf16 only work for torch version > 1.13
+                if compare_version("torch", operator.ge, "1.12.0"):
+                    # onednn fusion be added to torch from version 1.12
+                    torch.jit.enable_onednn_fusion(False)
+            # Disable AMP for JIT
+            torch._C._jit_set_autocast_mode(False)
```

Comment:
are we disabling bf16 when using "jit"?

Reply:
Not exactly.
_Currently, PyTorch has divergent AMP support in JIT & eager modes, so one should disable JIT AMP support & leverage eager mode AMP support to use BFloat16._
Here we just disable JIT AMP support and use eager mode AMP support.